File: site/assets/autres/primes.f90
PROGRAM primes

    PRINT*, "Test avec 131 : ", is_prime(131)
    PRINT*, "Test avec 25 : ", is_prime(25)

CONTAINS

    FUNCTION is_prime(n)
        LOGICAL :: is_prime
        INTEGER, INTENT(IN) :: n
        INTEGER :: k
    
        IF(n==2) THEN
        is_prime = .TRUE.
        ELSE IF(n < 2 .OR. MOD(n,2) == 0) THEN
        is_prime = .FALSE.
        ELSE
        is_prime = .TRUE.
        DO k = 3, INT(SQRT(REAL(n))) + 1, 2
            IF(MOD(n,k) == 0) THEN
                is_prime = .FALSE.
                EXIT
            END IF
        END DO
        END IF
    END FUNCTION is_prime

END PROGRAM primes

File: site/assets/autres/primes.ml
let is_prime n =
  if n = 2 then true
  else if n < 2 || n mod 2 = 0 then false
  else begin
    let reponse = ref true in 
    for k = 3 to 1 + int_of_float (sqrt (float_of_int n)) do
        if n mod k = 0 then
            reponse := false;
    done;
    !reponse;
    end;
;;

let str1 = string_of_bool (is_prime 131);;
print_string "Test avec 131 : ";;
print_string str1;;
print_newline ();;
let str2 = string_of_bool (is_prime 25);;
print_string "Test avec 25 : ";;
print_string str2;;